### DISCUSSION TITLE
Player/Output - DAC mixer information missing

### DISCUSSION THREAD

**Discussion by ceejayemm** [rern/rAudio#1622] [Bug Reports]
@rern 

I have been using a Pimoroni Audion DAC SHIM for some time with my Pi 3B+ test rig and although its a simple, cheap DAC it works well for me with rAudio for this purpose. The system is currently updated to 20240310. With recent updates though the DAC information seems to have disappeared from the Player  / Output page.  My setup is as shown below:

System Page:
![rac_system](https://github.com/rern/rAudio/assets/25418025/a57a7ca0-3d05-412f-a047-c81fe02e57c5)

Player Page:
![rac_player](https://github.com/rern/rAudio/assets/25418025/e2e9f5ef-cc3e-47b4-9e73-947ef75dafb0)

As you can see the Music Player Daemon appears to be paused and the Device and Mixer Device information on the Output section is missing. The 'log' section from Output reports:

# cat /proc/asound/cards | grep ]
 0 [sndrpihifiberry]: RPi-simple - snd_rpi_hifiberry_dac

# aplay -l | grep ^card
card 0: sndrpihifiberry [snd_rpi_hifiberry_dac], device 0: HifiBerry DAC HiFi pcm5102a-hifi-0 [HifiBerry DAC HiFi pcm5102a-hifi-0]

# amixer scontrols
(card 0: no mixers)

# cat /etc/asound.conf
defaults.pcm.card 0
defaults.ctl.card 0

Is this, now, expected behaviour given no mixers are being found ? I would have expected, in previous versions, to be able to choose between DAC Hardware and MPD Software options.  Music output seems unaffected, so is it defaulting to MPD Software in this case ?

Regards.

Chris

**Comment by rern** [comment]
Please post the output of: 
```sh
cat /proc/asound/card0/*/info

/srv/http/bash/settings/player-data.sh  | jq
```

**Reply by ceejayemm** [reply]
@rern 
As requested:

cat /proc/asound/card0/*/info
card: 0
device: 0
subdevice: 0
stream: PLAYBACK
id: bcm2835 Headphones
name: bcm2835 Headphones
subname: subdevice #0
class: 0
subclass: 0
subdevices_count: 8
subdevices_avail: 8

/srv/http/bash/settings/player-data.sh  | jq
{
  "page": "player",
  "login": false,
  "asoundcard": 1,
  "autoupdate": false,
  "bluetooth": false,
  "btmixer": "",
  "btvolume": false,
  "buffer": true,
  "bufferconf": {
    "KB": 8192
  },
  "camilladsp": false,
  "counts": {
    "album": 338,
    "albumartist": 131,
    "albumyear": 609,
    "artist": 648,
    "composer": 1052,
    "conductor": 2,
    "dabradio": 0,
    "date": 163,
    "genre": 38,
    "latest": 1,
    "nas": 1,
    "playlists": 0,
    "sd": 0,
    "song": 5113,
    "usb": 0,
    "webradio": 5
  },
  "crossfade": false,
  "crossfadeconf": {
    "SEC": 0
  },
  "custom": false,
  "dabradio": false,
  "devices": {
    "On-board Headphones": "hw:0,0",
    "HifiBerry DAC HiFi": "hw:1,0"
  },
  "devicewithbt": true,
  "dop": false,
  "equalizer": false,
  "ffmpeg": true,
  "lastupdate": "2024-02-21 <gr>• 10:31</gr>",
  "lists": {
    "albumignore": false,
    "mpdignore": true,
    "nonutf8": false
  },
  "mixers": false,
  "mixertype": false,
  "normalization": false,
  "output": {
    "card": 1,
    "name": "HifiBerry DAC HiFi",
    "mixer": false,
    "mixertype": "none"
  },
  "outputbuffer": false,
  "outputbufferconf": {
    "KB": 8192
  },
  "player": "mpd",
  "pllength": 1,
  "replaygain": false,
  "replaygainconf": {
    "MODE": "album",
    "HARDWARE": false
  },
  "soxr": false,
  "soxrconf": {
    "PLUGIN": "soxr",
    "QUALITY": "very high",
    "THREADS": 1
  },
  "soxrcustomconf": {
    "PLUGIN": "soxr",
    "QUALITY": "custom",
    "PRECISION": 20,
    "PHASE_RESPONSE": 50,
    "PASSBAND_END": 91.3,
    "STOPBAND_BEGIN": 100,
    "ATTENUATION": 0,
    "FLAGS": 0
  },
  "soxrquality": "",
  "state": "play",
  "updatetime": "(MPD: 2m 28s + Cache: 5s)",
  "updating_db": false,
  "version": "0.23.15-1",
  "volume": false
}

Thanks for the quick reply, I hope this helps.

Chris

**Reply by rern** [reply]
Hi Chris,

Try the UPDATE.

**Reply by ceejayemm** [reply]
@rern 
Applied latest Update branch.  No obvious change I am afraid.

Regards..

Chris
